tempesta_fw/ss_skb.c

```diff
@@ -816,6 +816,20 @@ __skb_fragment(struct sk_buff *skb_head, struct sk_buff *skb, char *pspt,
 			ret = __split_pgfrag(skb_head, skb, i, offset, len, it, fragn);
 			goto done;
 		}
+		/*
+		 * In case another fragment does not exist, remember the
+		 * fragment number, after which you need to insert a new one.
+		 */
+		if (offset == d_size) {
+			len = max_t(long, len, offset - d_size);
```

Comment:
Technically, we can be here with `len < 0`. There is no sense to delete `|len|` bytes after current fragment if there is no fragment right after the current one, but I'd prefer to have at least `BUG_ON()` assertion here or return with `-EINVAL` like:
```C
if (WARN_ON_ONCE(len < 0))
    return -EINVAL;
```